sentry-core/src/performance.rs

```diff
@@ -604,16 +604,26 @@ fn transaction_sample_rate(
 /// Determine whether the new transaction should be sampled.
 #[cfg(feature = "client")]
 impl Client {
-    fn is_transaction_sampled(&self, ctx: &TransactionContext) -> bool {
+    fn determine_sampling_decision(&self, ctx: &TransactionContext) -> (bool, f32) {
```

Comment:
[nit] Would add this doc comment so folks can see what is in the return value by just hovering over the function name in the IDE

```suggestion
    /// Returns sampling decision and sample rate based on the transaction context
    fn determine_sampling_decision(&self, ctx: &TransactionContext) -> (bool, f32) {
```